### PR TITLE
DOCSUP-6258: FAQ edit and translate

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #11898

### CODE REVIEW

File: docs/ru/faq/general/mapreduce.md [hunk 1]
```diff
@@ -6,11 +6,8 @@ toc_priority: 110
...
-Распределённая сортировка является основной причиной тормозов при выполнении несложных map-reduce задач.
+Такие системы не подходят для онлайн запросов в силу слишком большой задержки. То есть, они не подходят в качестве бэкенда для веб-интерфейса. Также эти системы не подходят для обновления данных в реальном времени. Распределённая сортировка не оптимальный способ для выполнения операции свёртки, если результат выполнения операции и все промежуточные результаты (если такие есть), помещаются в оперативку на одном сервере, как обычно бывает в запросах, выполняющихся в режиме онлайн. В таком случае оптимальным способом выполнения операции свёртки становится хеш-таблица. Частым способом оптимизации "map-reduce" задач является предагрегация (частичная свёртка) с использованием хеш-таблицы в оперативной памяти. Пользователь делает эту оптимизацию в ручном режиме. Распределённая сортировка — основная причина тормозов при выполнении несложных задач типа "map-reduce".
```
Comment (adevyatova) [inline docs/ru/faq/general/mapreduce.md:11]:
```suggestion
Такие системы не подходят для онлайн запросов в силу слишком большой задержки. То есть, не могут быть использованы в качестве бэкенда для веб-интерфейса. Также эти системы не подходят для обновления данных в реальном времени. Распределённая сортировка является не оптимальным способом для выполнения операции свёртки в случае запросов, выполняющихся в режиме онлайн, потому что результат выполнения операции и все промежуточные результаты (если такие есть), помещаются в оперативку на одном сервере. В таком случае оптимальным способом выполнения операции свёртки является хеш-таблица. Частым способом оптимизации "map-reduce" задач является предагрегация (частичная свёртка) с использованием хеш-таблицы в оперативной памяти. Пользователь делает эту оптимизацию в ручном режиме. Распределённая сортировка — основная причина тормозов при выполнении несложных задач типа "map-reduce".
```

File: docs/ru/faq/general/dbms-naming.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+toc_priority: 10
+---
+
+# Что означает название ClickHouse? {#what-does-clickhouse-mean}
+
+Это комбинация терминов **Click**stream и Data ware**House**. Название пришло из Яндекс.Метрики, для которой первоначально был разработан ClickHouse — там он использовался для хранения истории визитов пользователей на сайты и всех пользовательских действий — "кликов". Кстати, ClickHouse по-прежнему выполняет эту функцию. Узнать об этом больше можно на странице [истории ClickHouse](../../introduction/history.md).
+
+Поскольку название составное, использовать его нужно следующим образом:
+
+-   Единственно правильный способ написания — Click**H**ouse — с заглавной буквой H.
```
Comment (adevyatova) [inline docs/ru/faq/general/dbms-naming.md:13]:
```suggestion
-   единственно правильный способ написания — Click**H**ouse — с заглавной буквой H.
```

File: docs/ru/faq/general/dbms-naming.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+-   Если нужно сокращеннное название, используйте **CH**. Исторически сложилось, что в Китае также популярно сокращение CK — в основном, из-за того, что это название использовалось в одном из первых обсуждений ClickHouse на китайском языке.
```
Comment (adevyatova) [inline docs/ru/faq/general/dbms-naming.md:14]:
```suggestion
-   если нужно сокращеннное название, используйте **CH**. Исторически сложилось, что в Китае также популярно сокращение CK — в основном, из-за того, что это название использовалось в одном из первых обсуждений ClickHouse на китайском языке.
```

File: docs/ru/faq/general/ne-tormozit.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+-   Если переводить буквально, то получится что-то подобное: *“ClickHouse doesn’t press the brake pedal”*.
+-   Если же вы хотите максимально сохранить том смысл, который вкладывает в эту фразу человек из ИТ-сферы, то будет примерно следующее: *“If your larger system lags, it’s not because it uses ClickHouse”*. 
+-   Более короткие, но не такие точные версии: *“ClickHouse is not slow”*, *“ClickHouse doesn’t lag”* или просто *“ClickHouse is fast”*.
+
+Если вы не видели наших футболок, посмотрите видео о ClickHouse. Например, вот это:
+
+![iframe](https://www.youtube.com/embed/bSyQahMVZ7w)
+
+P.S. Эти футболки не продаются, а распространяются бесплатно на большинстве митапах [ClickHouse](https://clickhouse.tech/#meet), обычно в награду за самые интересные вопросы или другие виды активного участия.
```
Comment (adevyatova) [inline docs/ru/faq/general/ne-tormozit.md:26]:
```suggestion
P.S. Эти футболки не продаются, а распространяются бесплатно на большинстве митапов [ClickHouse](https://clickhouse.tech/#meet), обычно в награду за самые интересные вопросы или другие виды активного участия.
```

File: docs/ru/faq/general/olap.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+На практике OLAP и OLTP это не строго разделённые категории, а скорее спектр возможностей. Большинство СУБД специализируются на каком-то одном виде обработки данных, но имеют инструменты и для выполнения других операций, когда это необходимо. Из-за такой специализации часто приходится использовать несколько СУБД и интегрировать их между собой. Это вполне реальная и решаемая задача, но как известно, чем больше систем, тем выше расходы на их содержание. Поэтому в последние годы становятся популярны гибридные СУБД — HTAP (**Hybrid Transactional/Analytical Processing**), которые одинаково эффективно выполняют оба вида операций по обработке данных.
+
+Даже если СУБД сначала развивались исключительно как OLAP или как OLTP, постепеннл двигаются в сторону HTAP, чтобы сохранять конкурентоспособность. И ClickHouse не исключение. Изначально он создавался как [OLAP СУБД с максимальной производительностью](../../faq/general/why-clickhouse-is-so-fast.md), и на сегодняшний день в нем нет полноценной поддержки обработки тразакций, но уже реализованы некоторые возможности, такие как постоянная скорость чтения/записи данных и мутации при изменении/удалении данных.
```
Comment (adevyatova) [inline docs/ru/faq/general/olap.md:34]:
```suggestion
Даже если СУБД сначала развивались исключительно как OLAP или как OLTP, разработчики постепенно двигаются в сторону HTAP, чтобы сохранять конкурентоспособность. И ClickHouse не исключение. Изначально он создавался как [OLAP СУБД с максимальной производительностью](../../faq/general/why-clickhouse-is-so-fast.md), и на сегодняшний день в нем нет полноценной поддержки обработки тразакций, но уже реализованы некоторые возможности, такие как постоянная скорость чтения/записи данных и мутации при изменении/удалении данных.
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
+---
+title: Почему ClickHouse так быстро работает?
+toc_hidden: true
+toc_priority: 8
+---
+
+#  Почему ClickHouse так быстро работает? {#why-clickhouse-is-so-fast}
+
+Производительность изначально заложена в архитектуре ClickHouse. Высокая скорость выполнения запросов была и остается самым важным критерием, который учитывается при разработке. Но мы обращаем внимание и на другие характеристики, такие как удобство использования, масштабируемость, безопасность. Всё это делает ClickHouse настоящей промышленной разработкой.
```
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:9]:
```suggestion
Производительность изначально заложена в архитектуре ClickHouse. Высокая скорость выполнения запросов была и остается самым важным критерием, который учитывается при разработке. Но мы обращаем внимание и на другие характеристики, такие как: удобство использования, масштабируемость, безопасность. Всё это делает ClickHouse настоящей промышленной разработкой. 
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+title: Почему ClickHouse так быстро работает?
+toc_hidden: true
+toc_priority: 8
+---
+
+#  Почему ClickHouse так быстро работает? {#why-clickhouse-is-so-fast}
+
+Производительность изначально заложена в архитектуре ClickHouse. Высокая скорость выполнения запросов была и остается самым важным критерием, который учитывается при разработке. Но мы обращаем внимание и на другие характеристики, такие как удобство использования, масштабируемость, безопасность. Всё это делает ClickHouse настоящей промышленной разработкой. 
+
+Сначала ClickHouse создавали как прототип, который должен был отлично справляться с одной единственной задачей — отбирать и агрегировать данные с максимальной скоростью. Именно это необходимо, чтобы создать обычный аналитический отчет, и именно это делает стандартный запрос [GROUP BY](../../sql-reference/statements/select/group-by.md). Чтобы добиться этого, команда разработки ClickHouse приняла несколько архитектурных решений:
```
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:11]:
```suggestion
Сначала ClickHouse создавался как прототип, который должен был отлично справляться с одной единственной задачей — отбирать и агрегировать данные с максимальной скоростью. Это необходимо, чтобы создать обычный аналитический отчет, и именно это делает стандартный запрос [GROUP BY](../../sql-reference/statements/select/group-by.md). Чтобы добиться этого, команда разработки ClickHouse приняла несколько архитектурных решений: 
```

File: docs/ru/faq/general/olap.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+## OLAP с технической точки зрения {#olap-from-the-technical-perspective}
+
+Все СУБД можно разделить на две группы: OLAP (**аналитическая** обработка в реальном времени) и OLTP (обработка **транзакций** в реальном времени). OLAP используются для построения отчетов на основе больших объемов накопленных исторических данных, но эти отчеты обновляются не слишком часто. OLTP обычно применяются для обработки непрерывных потоков операций (транзакций), каждая из которых изменяет состояние данных.
+
+На практике OLAP и OLTP это не строго разделённые категории, а скорее спектр возможностей. Большинство СУБД специализируются на каком-то одном виде обработки данных, но имеют инструменты и для выполнения других операций, когда это необходимо. Из-за такой специализации часто приходится использовать несколько СУБД и интегрировать их между собой. Это вполне реальная и решаемая задача, но как известно, чем больше систем, тем выше расходы на их содержание. Поэтому в последние годы становятся популярны гибридные СУБД — HTAP (**Hybrid Transactional/Analytical Processing**), которые одинаково эффективно выполняют оба вида операций по обработке данных.
```
Comment (adevyatova) [inline docs/ru/faq/general/olap.md:32]:
```suggestion
На практике OLAP и OLTP — это не строго разделённые категории, а скорее спектр возможностей. Большинство СУБД специализируются на каком-то одном виде обработки данных, но имеют инструменты и для выполнения других операций, когда это необходимо. Из-за такой специализации часто приходится использовать несколько СУБД и интегрировать их между собой. Это вполне реальная и решаемая задача, но как известно, чем больше систем, тем выше расходы на их содержание. Поэтому в последние годы становятся популярны гибридные СУБД — HTAP (**Hybrid Transactional/Analytical Processing**), которые одинаково эффективно выполняют оба вида операций по обработке данных.
```

File: docs/ru/faq/general/columnar-database.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+toc_priority: 101
+---
+
+# Что такое столбцовая (колоночная) база данных? {#what-is-a-columnar-database}
+
+В столбцовой БД данные каждого столбца хранятся отдельно (независимо) от других столбцов. Такой принцип хранения позволяет при выполнении запроса считывать с диска данные только тех столбцов, которые непосредственно участвуют в этом запросе. Обратная сторона такого принципа хранения заключается в том, что выполнение операций над строками становится более затратным. ClickHouse — типичный пример столбцовой СУБД.
+
+Ключевые преимущества столбцовой СУБД: 
+
+- Выполнение запросов над отдельными столбцами таблицы, а не над всей таблицей сразу.
```
Comment (adevyatova) [inline docs/ru/faq/general/columnar-database.md:13]:
```suggestion
- выполнение запросов над отдельными столбцами таблицы, а не над всей таблицей сразу.
```

File: docs/ru/faq/general/columnar-database.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+- Агрегация запросов на больших объемах данных.
```
Comment (adevyatova) [inline docs/ru/faq/general/columnar-database.md:14]:
```suggestion
- агрегация запросов на больших объемах данных. 
```

File: docs/ru/faq/general/columnar-database.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+# Что такое столбцовая (колоночная) база данных? {#what-is-a-columnar-database}
+
+В столбцовой БД данные каждого столбца хранятся отдельно (независимо) от других столбцов. Такой принцип хранения позволяет при выполнении запроса считывать с диска данные только тех столбцов, которые непосредственно участвуют в этом запросе. Обратная сторона такого принципа хранения заключается в том, что выполнение операций над строками становится более затратным. ClickHouse — типичный пример столбцовой СУБД.
+
+Ключевые преимущества столбцовой СУБД: 
+
+- Выполнение запросов над отдельными столбцами таблицы, а не над всей таблицей сразу.
+- Агрегация запросов на больших объемах данных. 
+- Сжатие данных в столбцах.
```
Comment (adevyatova) [inline docs/ru/faq/general/columnar-database.md:15]:
```suggestion
- сжатие данных в столбцах.
```

File: docs/ru/faq/general/dbms-naming.md (hunk 2 above)
Comment (adevyatova) [inline docs/ru/faq/general/dbms-naming.md:13]:
```suggestion
-   единственно правильный способ написания — Click**H**ouse — с заглавной буквой H.
```

File: docs/ru/faq/general/dbms-naming.md (hunk 3 above)
Comment (adevyatova) [inline docs/ru/faq/general/dbms-naming.md:14]:
```suggestion
-   если нужно сокращеннное название, используйте **CH**. Исторически сложилось, что в Китае также популярно сокращение CK — в основном, из-за того, что это название использовалось в одном из первых обсуждений ClickHouse на китайском языке.
```

File: docs/ru/faq/general/ne-tormozit.md (hunk 4 above)
Comment (adevyatova) [inline docs/ru/faq/general/ne-tormozit.md:26]:
```suggestion
P.S. Эти футболки не продаются, а распространяются бесплатно на большинстве митапов [ClickHouse](https://clickhouse.tech/#meet), обычно в награду за самые интересные вопросы или другие виды активного участия.
```

File: docs/ru/faq/general/olap.md (hunk 5 above)
Comment (adevyatova) [inline docs/ru/faq/general/olap.md:34]:
```suggestion
Даже если СУБД сначала развивались исключительно как OLAP или как OLTP, разработчики постепенно двигаются в сторону HTAP, чтобы сохранять конкурентоспособность. И ClickHouse не исключение. Изначально он создавался как [OLAP СУБД с максимальной производительностью](../../faq/general/why-clickhouse-is-so-fast.md), и на сегодняшний день в нем нет полноценной поддержки обработки тразакций, но уже реализованы некоторые возможности, такие как постоянная скорость чтения/записи данных и мутации при изменении/удалении данных.
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md (hunk 6 above)
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:9]:
```suggestion
Производительность изначально заложена в архитектуре ClickHouse. Высокая скорость выполнения запросов была и остается самым важным критерием, который учитывается при разработке. Но мы обращаем внимание и на другие характеристики, такие как: удобство использования, масштабируемость, безопасность. Всё это делает ClickHouse настоящей промышленной разработкой. 
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md (hunk 7 above)
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:11]:
```suggestion
Сначала ClickHouse создавался как прототип, который должен был отлично справляться с одной единственной задачей — отбирать и агрегировать данные с максимальной скоростью. Именно это необходимо, чтобы создать обычный аналитический отчет, и именно это делает стандартный запрос [GROUP BY](../../sql-reference/statements/select/group-by.md). Чтобы добиться этого, команда разработки ClickHouse приняла несколько архитектурных решений: 
```

File: docs/ru/faq/operations/production.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+Во-первых, давайте обсудим, почему люди задают этот вопрос сразу. Две основные причины:
+
+1.  ClickHouse разработали достаточно быстро и обычно есть более 10-ти стабильных релизов в год. Так что есть из чего выбирать, а это не всегда просто. 
+2.  Некоторые пользователи хотят избежать трат времени на выяснение, какая версия работает лучше для их сценария использования и просто хотят послушать толковый совет.
+
+Вторая причина более весомая, так что начнем с нее и затем вернемся к навигации по существующим релизам ClickHouse.
+
+## Какую версию ClickHouse вы посоветуете? {#which-clickhouse-version-do-you-recommend}
+
+Очень удобный вариант — нанять консультанта или довериться известному эксперту, чтобы делегировать ответственность за вашу производственную среду. Вы устанавливаете одну из верси ClickHouse, которую порекомендовал кто-то, а теперь если с ней что-то идет не так — это уже не ваша вина, а тех, кто давал совет. Такая линия причинно-следственной связи очень большая ловушка. Никто другой, кроме вас не знает, что происходит в производственной среде вашей компании.
```
Comment (adevyatova) [inline docs/ru/faq/operations/production.md:18]:
```suggestion
Очень удобный вариант — нанять консультанта или довериться известному эксперту, чтобы делегировать ответственность за вашу производственную среду. Вы устанавливаете одну из версий ClickHouse, которую порекомендовал кто-то, а теперь если с ней что-то идет не так — это уже не ваша вина, а тех, кто давал совет. Такая линия причинно-следственной связи очень большая ловушка. Никто другой, кроме вас не знает, что происходит в производственной среде вашей компании. 
```

File: docs/ru/faq/general/olap.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+
+## OLAP с технической точки зрения {#olap-from-the-technical-perspective}
+
+Все СУБД можно разделить на две группы: OLAP (**аналитическая** обработка в реальном времени) и OLTP (обработка **транзакций** в реальном времени). OLAP используются для построения отчетов на основе больших объемов накопленных исторических данных, но эти отчеты обновляются не слишком часто. OLTP обычно применяются для обработки непрерывных потоков операций (транзакций), каждая из которых изменяет состояние данных.
+
+На практике OLAP и OLTP — это не строго разделённые категории, а скорее спектр возможностей. Большинство СУБД специализируются на каком-то одном виде обработки данных, но имеют инструменты и для выполнения других операций, когда это необходимо. Из-за такой специализации часто приходится использовать несколько СУБД и интегрировать их между собой. Это вполне реальная и решаемая задача, но как известно, чем больше систем, тем выше расходы на их содержание. Поэтому в последние годы становятся популярны гибридные СУБД — HTAP (**Hybrid Transactional/Analytical Processing**), которые одинаково эффективно выполняют оба вида операций по обработке данных.
```
Comment (adevyatova) [inline docs/ru/faq/general/olap.md:32]:
```suggestion
На практике OLAP и OLTP — это не строго разделённые категории, а скорее спектр возможностей. Большинство СУБД специализируются на каком-то одном виде обработки данных, но имеют инструменты и для выполнения других операций, когда это необходимо. Из-за такой специализации часто приходится использовать несколько СУБД и интегрировать их между собой. Это вполне реальная и решаемая задача, но, как известно, чем больше систем, тем выше расходы на их содержание. Поэтому в последние годы становятся популярны гибридные СУБД — HTAP (**Hybrid Transactional/Analytical Processing**), которые одинаково эффективно выполняют оба вида операций по обработке данных.
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+Векторные запросы
+:   ClickHouse не только хранит, но и обрабатывает данные в столбцах. Это приводит к лучшей утилизации кеша процессора и позволяет использовать инструкции [SIMD](https://en.wikipedia.org/wiki/SIMD).
+
+Масштабируемость
+:   ClickHouse может задействовать все доступные мощности процессоров и объемы дисков, чтобы выполнить даже одиночный запрос. Не только на отдельном сервере, но и в целом кластере.
+
+Похожие техники используют и многие другие СУБД. **Внимание к мельчайшим деталям** — вот что на самом деле выделяет ClickHouse. Большинство языков программирования поддерживают большинство распространенных алгоритмов и структур данных, но как правило они бывают слишком универсальными, чтобы быть по-настоящему эффективными. Мы рассматриваем каждую задачу как тонкий инструмент со множеством настроек, вместо того чтобы просто взять какую-то случайную реализацию. Например, если вам нужна хеш-таблица, вот несколько ключевых вопросов, которые нужно продумать:
```
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:28]:
```suggestion
Похожие техники используют и многие другие СУБД. **Внимание к мельчайшим деталям** — вот что на самом деле выделяет ClickHouse. Большинство языков программирования поддерживают большинство распространенных алгоритмов и структур данных, но как правило, они бывают слишком универсальными, чтобы быть по-настоящему эффективными. Мы рассматриваем каждую задачу как тонкий инструмент со множеством настроек, вместо того чтобы просто взять какую-то случайную реализацию. Например, если вам нужна хеш-таблица, вот несколько ключевых вопросов, которые нужно продумать:
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+Для алгоритмов сортировки, например, следует продумать следующие вопросы:
+
+-   Что будет сортироваться: массив чисел, кортежей, строк или структур?
+-   Все ли данные доступны в оперативной памяти?
+-   Нужна ли стабильная сортировка?
+-   Нужна ли нам полная сортировка? Может быть, будет достаточно частичной или выборочной сортировки?
+-   Как сравнивать данные?
+-   Сортируем ли мы данные, которые уже частично отсортированы?
+
+Алгоритмы, которые основываются на характеристиках данных, с которыми работают, обычно дают лучшие результаты, чем их более универсальные аналоги. Если заранее неизвестно, с какими данными придется работать, ClickHouse будет в процессе выполнения пробовать различные реализации и в итоге выберет оптимальный вариант. Например, рекомендуем прочитать [статью о том, как в ClickHouse реализуется распаковка LZ4](https://habr.com/en/company/yandex/blog/457612/).
```
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:48]:
```suggestion
Алгоритмы, основанные на характеристиках данных, с которыми работают, обычно дают лучшие результаты, чем их более универсальные аналоги. Если заранее неизвестно, с какими данными придется работать, ClickHouse будет в процессе выполнения пробовать различные реализации и в итоге выберет оптимальный вариант. Например, рекомендуем прочитать [статью о том, как в ClickHouse реализуется распаковка LZ4](https://habr.com/en/company/yandex/blog/457612/). 
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+
+    -   При проектировании системы обращайте внимание на мельчайшие детали реализации.
+    -   Учитывайте возможности аппаратного обеспечения.
+    -   Выбирайте структуры и представления данных исходя из требований конкретной задачи.
+    -   Для особых случаев разрабатывайте специализированные решения.
+    -   Пробуйте новые алгоритмы, о которых вы вчера прочитали в интернете. Ищите возможности для совершенствования.
+    -   Выбирайте алгоритмы динамически, в процессе выполнения, на основе статистики.
+    -   Ориентируйтесь на показатели, собранные при работе с реальными данными.
+    -   Проверяйте производительность в процессе CI.
```
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:62]:
```suggestion
    -   В процессе непрерывной интеграции проверяйте производительность.
```

File: docs/ru/faq/general/why-clickhouse-is-so-fast.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+-   Понадобится ли нам быстрое зондирование с bitmaps, вставка строчных ключей, поддержка неперемещаемых значений, предварительная выборка и пакетирование?
+
+Хеш-таблица — ключевая структура данных для реализации `GROUP BY`, и ClickHouse автоматически выбирает одну из [более 30 вариаций](https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/Aggregator.h) для каждого специфического запроса.
+
+Для алгоритмов сортировки, например, следует продумать следующие вопросы:
+
+-   Что будет сортироваться: массив чисел, кортежей, строк или структур?
+-   Все ли данные доступны в оперативной памяти?
+-   Нужна ли стабильная сортировка?
+-   Нужна ли нам полная сортировка? Может быть, будет достаточно частичной или выборочной сортировки?
```
Comment (adevyatova) [inline docs/ru/faq/general/why-clickhouse-is-so-fast.md:44]:
```suggestion
-   Нужна ли полная сортировка? Может быть, будет достаточно частичной или выборочной сортировки?
```

**Review by vdimir — APPROVED** [review]
Thank you! I'm going to commit minor changes and merge this PR.

**Pull request merged**